src/stan/lang/ast/sigs/function_signatures_def.hpp

```diff
@@ -175,34 +175,39 @@ namespace stan {
     }
 
     void function_signatures::add_nullary(const::std::string& name) {
-      add(name, DOUBLE_T);
+      add(name, expr_type(double_type()));
```

Comment:
This is where the implicit construtors you have could just let you use `double_type()`.  If you mark the constructors as `explicit`, then you need to wrap in `expr_type`.  The explicit approach is usually better in terms of there being fewer unintended consequences (like promotions where you don't expect them).

Reply:
leaving as is for code clarity.